client/lib/user-settings/index.js

```diff
@@ -29,6 +26,26 @@ function decodeUserSettingsEntities( data ) {
 	return assign( {}, data, decodedValues );
 }
 
+/*
+ * Deletes a provided unsaved setting, then calls itself recursively
+ * to delete any empty parents of the setting passed to it
+ */
+function deleteUnsavedSetting( settings, settingName, recursive ) {
+	if ( ! isEmpty( get( settings, settingName ) ) || recursive ) {
+		unset( settings, settingName );
+
+		const settingKeys = settingName.split( '.' );
+		if ( settingKeys.length > 1 ) {
```

Comment:
Ahh my bad! For some reason I was reading/treating this as `settingKeys.length >= 1`!